### PR TITLE
feature/24

### LINKED ISSUE #24: Command to inform arrivals of the arrival runway to expect
Need a command we can use to inform an aircraft of what runway they can expect for arrival. This would be used primarily to help them determine which branches of a given STAR they should be following, as there are often different paths to follow in the chart based on the runway they are to land on.

See zlsa/atc#500.

### PR DESCRIPTION
**Pull request by erikquinn**
Resolves #24.

Add command to inform arrivals of assigned landing runway.

### CODE REVIEW

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js [hunk 1]
```diff
@@ -323,7 +288,12 @@ export default class AircraftCommandParser {
...
     _isSystemCommand(callsignOrSystemCommandName) {
-        return _has(SYSTEM_COMMANDS, callsignOrSystemCommandName) &&
-            callsignOrSystemCommandName !== SYSTEM_COMMANDS.transmit;
+        const command = AIRCRAFT_COMMAND_MAP[callsignOrSystemCommandName];
+
+        if (typeof command === 'undefined') {
+            return false;
+        }
+
+        return command.isSystemCommand && callsignOrSystemCommandName !== 'transmit';
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: `'transmit'` should be an enum

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandDefinitions.js [hunk 2]
```diff
@@ -75,7 +75,7 @@ const ZERO_ARG_AIRCRAFT_COMMANDS = {
         validate: zeroArgumentsValidator,
         parse: noop
     },
-    climbViaSID: {
+    climbViaSid: {
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandDefinitions.js]: 👍 

I've never been a fan of the `climbViaSID` capitalization.  thank you for changing this!

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js [hunk 3]
```diff
@@ -1,40 +1,6 @@
...
- * @property UNICODE_COMMANDS
- * @type {Object}
- * @final
- */
-const UNICODE_COMMANDS = {
-    '\\u2B50': 'land'
-};
+import _findKey from 'lodash/findKey';
 
+// FIXME: These keys should be upper-cased, but waiting for now to keep things working
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: why should they be upper-cased?  since these map directly to method names, I'd say it makes sense for them to be camelCase.  We should probably have a comment as to _why_ they are camelCased, though.

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js [hunk 4]
```diff
@@ -54,67 +20,191 @@ const UNICODE_COMMANDS = {
...
+        functionName: 'runHeading',
+        isSystemCommand: false
+    },
+    hold: {
+        aliases: ['hold'],
+        functionName: 'runHold',
+        isSystemCommand: false
+    },
+    land: {
+        aliases: ['\\u2B50', '*', 'i', 'ils'],
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: are we still handling unicode commands?

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js [hunk 5]
```diff
@@ -123,3 +213,14 @@ export const AIRCRAFT_COMMAND_MAP = {
...
 export const EXPEDITE = ['expedite', 'x'];
+
+/**
+ * Get the name of a command when given any of that command's aliases
+ *
+ * @function getCommandNameFromAlias
+ * @param alias {string}
+ * @return {string}
+ */
+export function getCommandNameFromAlias(alias) {
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: `.findCommandNameWithAlias()`

and could we change the param name to: `commandAlias`

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js [hunk 6]
```diff
@@ -54,67 +20,191 @@ const UNICODE_COMMANDS = {
  * @final
  */
 export const AIRCRAFT_COMMAND_MAP = {
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: nice work here!  I really like the simplification you've done here.

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 4 above)
Comment (sam-irl) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: No; I believe if an arrow key is pressed it's replaced with `c` or `d` in the command bar. So a Unicode character would never be sent to be parsed.

Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: this is the star that's inserted when pressing the numpad `*`. I'm not sure if it's in use, was one of the loose ends to be cleaned up when I get to it. Still a work in progress of course.

Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: it was part of a `UNICODE_COMMANDS` enum that was `...`ed into the `AIRCRAFT_COMMANDS_MAP`. so i haven't changed anything really, but nonetheless intend to find out if it should be removed or not.

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 5 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: works for me. It's a helper function though-- you don't mean to add it to a class or anything do you? Just since you used a dot in front.

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js [hunk 7]
```diff
@@ -122,7 +119,7 @@ export default class AircraftCommandParser {
      * @return {string|array<string>}
      */
     get args() {
-        if (this.command !== SYSTEM_COMMANDS.transmit) {
+        if (this.command !== 'transmit') {
```
[the remark this answers is not in the document]
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: that's the plan

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 5 above)
Comment (n8rzz) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: Oh, no, it's just fine as a function.

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: done

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js [hunk 8]
```diff
@@ -187,7 +184,7 @@ export default class AircraftCommandParser {
      * @private
      */
     _buildTransmitAircraftCommandModels(callsignOrSystemCommandName, commandArgSegments) {
-        this.command = SYSTEM_COMMANDS.transmit;
+        this.command = 'transmit';
```
[the remark this answers is not in the document]
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: done

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js (hunk 1 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: done

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandDefinitions.js (hunk 2 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandDefinitions.js]: yeah, it used to make me _crazy_ to see `Sid` because that's like seeing `Usa`: it's just wrong. But I've come around to the convention of keeping _EVERYTHING_ proper camel cased. Makes it easy and consistent, so here we are. 😃

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 3 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: Kinda just because it's been the pattern. If it doesn't bother you, I'll leave it because frankly I've grown tired of this PR anyway. 😉

Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: turns out there was already a comment to this effect, so i simply removed the fixme.

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 6 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: 🎉

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 4 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: not really needed. removed, and redirected the hotkey to insert the `*` alias.

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 5 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: done

**Comment by erikquinn** [comment]
likely to be done as part of the route/fms refactor underway.

**Comment by erikquinn** [comment]
not done under that refactor. Will need to reassess/restart/resume this at some point.

File: src/assets/scripts/client/InputController.js [hunk 9]
```diff
@@ -406,7 +406,7 @@ export default class InputController {
 
                 break;
             case KEY_CODES.MULTIPLY:
-                this.$commandInput.val(`${currentCommandInputValue} \u2B50 `);
+                this.$commandInput.val(`${currentCommandInputValue} * `);
```
Comment (n8rzz) [inline src/assets/scripts/client/InputController.js]: why the change here?  

remember, the `*` is interpreted differently by operating system.  the `*` on windows is not the same as a `*` on osx, hence the unicode 😉

File: src/assets/scripts/client/aircraft/AircraftCommander.js [hunk 10]
```diff
@@ -361,6 +316,34 @@ export default class AircraftCommander {
...
+     * @param data {array}
+     * @return {array} [success of operation, response]
+     */
+    runExpectArrivalRunway(aircraft, data) {
+        const airportModel = AirportController.airport_get();
+        const runwayName = data[0];
+        const runwayModel = airportModel.getRunway(runwayName);
+
+        if (_isNil(runwayModel)) {
+            const oldRunwayModel = aircraft.fms.arrivalRunwayModel;
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftCommander.js]: what does `old` mean here?   previous, perhaps?

File: src/assets/scripts/client/aircraft/AircraftCommander.js [hunk 11]
```diff
@@ -592,19 +575,23 @@ export default class AircraftCommander {
...
+        const runwayModel = AirportController.airport_get().getRunway(taxiDestination.toUpperCase());
+
+        if (!runwayModel) {
+            return [false, `unable to find Runway ${taxiDestination.toUpperCase()} on our charts`];
+        }
 
-        if (!runway) {
-            return [false, `no runway ${taxiDestination.toUpperCase()}`];
+        if (!aircraft.fms.isRunwayModelValidForSid(runwayModel)) {
+            aircraft.pilot.cancelDepartureClearance(aircraft);
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftCommander.js]: does `.cancelDepartureClearance()` actually need the `aircraft` param?  seems a little weird, since it's the same instance calling that method.

File: src/assets/scripts/client/aircraft/AircraftCommander.js [hunk 12]
```diff
@@ -592,19 +575,23 @@ export default class AircraftCommander {
...
         }
 
-        const readback = aircraft.pilot.taxiToRunway(runway, isDeparture, flightPhase);
+        const readback = aircraft.pilot.taxiToRunway(runwayModel, isDeparture, flightPhase);
 
         // TODO: this may need to live in a method on the aircraft somewhere
-        aircraft.fms.departureRunwayModel = runway;
+        aircraft.fms.departureRunwayModel = runwayModel;
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftCommander.js]: do we have a method in the `Fms` to update the runway?  I am wary about setting things like this outside the class, it makes it hard to track things down.

I understand it was already like this, buuuuuut, now's the time to tackle these.  Or maybe just update the comment, this method should live in the `Fms`.

File: src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js [hunk 13]
```diff
@@ -636,6 +636,30 @@ export default class Fms {
         this.currentPhase === FLIGHT_PHASE.CLIMB;
     }
 
+    /**
+     * Facade for #_routeModel.isRunwayModelValidForSid()
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js]: missing bat ticks around `#_routeModel.isRunwayModelValidForSid()`

Comment (n8rzz) [reply src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js]: could we also add a blurb about _why_ this facade is needed?  for this and the next one below please.

File: src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js [hunk 14]
```diff
@@ -638,6 +638,54 @@ export default class RouteModel extends BaseModel {
...
+     * Returns whether the specified runway is valid for this route's STAR leg (if it has one)
+     *
+     * If there is no STAR, there is no issue with changing runways, so we would treat this as "valid"
+     *
+     * @for RouteModel
+     * @method isRunwayModelValidForStar
+     * @param runwayModel {RunwayModel}
+     * @return {boolean}
+     */
+    isRunwayModelValidForStar(runwayModel) {
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js]: I'd encourage you to see where these two methods could share logic.  there is a fair amount of repeated code between these two methods.

File: src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js [hunk 15]
```diff
@@ -785,29 +834,51 @@ export default class RouteModel extends BaseModel {
...
+            const procedureModelName = starLegModel.getProcedureName();
+            const readback = {};
+            readback.log = `unable, Runway ${runwayModel.name} is not valid for the ${procedureModelIcao} arrival`;
+            readback.say = `unable, Runway ${runwayModel.name} is not valid for the ${procedureModelName} arrival`;
+
+            return [false, readback];
+        }
+
+        const amendedStarLegModel = this._createAmendedStarLegUsingDifferentExitName(nextExitName, starLegIndex);
+
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js]: remove empty line

File: src/assets/scripts/client/aircraft/Pilot/Pilot.js [hunk 16]
```diff
@@ -381,7 +399,20 @@ export default class Pilot {
      * @param {Number} initialAltitude  the altitude aircraft can automatically climb to at this airport
      * @return {Array}                  [success of operation, readback]
      */
-    clearedAsFiled() {
+    clearedAsFiled(aircraft) {
+        const fms = aircraft.fms;
+        const departureRunwayModel = fms.departureRunwayModel;
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/Pilot/Pilot.js]: `fms` is unnecessary here.  instead try:
```js
const { departureRunwayModel } = fms;
```

This also seems overly defensive.  We're checking that the `departureRunwayModel` that we've already set is valid?  Shouldn't that happen when we're setting it initially and not here?  This method should really just be like: _'Ok, we have your flight plan, we verified it on instantiation, go ahead and fly'_

Comment (n8rzz) [reply src/assets/scripts/client/aircraft/Pilot/Pilot.js]: I'm all for defensive programming, it just feels unnecessary here.

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js [hunk 17]
```diff
@@ -65,7 +63,7 @@ export default class AircraftCommandParser {
      * @for AircraftCommandParser
      * @param rawCommandWithArgs {string}  string present in the `$commandInput` when the user pressed `enter`
      */
-    constructor(rawCommandWithArgs = '') {
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: is this defaulted somewhere else?  why remove the default here?

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js [hunk 18]
```diff
@@ -299,7 +275,11 @@ export default class AircraftCommandParser {
      * @private
      */
     _validateCommandArguments() {
-        return _compact(_map(this.commandList, (command) => {
+        const validatedCommandList = _map(this.commandList, (command) => {
+            if (typeof command === 'undefined') {
+                return;
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: this should still return a falsey value like `null` because using the `compact` below will remove those values properly.  just makes things more clear.  yay verbosity!

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js [hunk 19]
```diff
@@ -54,67 +19,191 @@ const UNICODE_COMMANDS = {
...
+        functionName: 'runHeading',
+        isSystemCommand: false
+    },
+    hold: {
+        aliases: ['hold'],
+        functionName: 'runHold',
+        isSystemCommand: false
+    },
+    land: {
+        aliases: ['*', 'i', 'ils'],
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: be careful with the `*` again here.  this may not be interpreted the same across operating systems

File: test/aircraft/FlightManagementSystem/RouteModel.spec.js [hunk 20]
```diff
@@ -738,7 +738,7 @@ ava('.updateSidLegForDepartureRunwayModel() calls LegModel.updateStarLegForArriv
     t.true(sidLegUpdateSidRunwaySpy.calledWithExactly(nextRunwayModel));
 });
 
-ava('.updateStarLegForArrivalRunwayModel() returns early when the route contains no STAR leg to update', (t) => {
+ava.skip('.updateStarLegForArrivalRunwayModel() returns early when the route contains no STAR leg to update', (t) => {
```
Comment (n8rzz) [inline test/aircraft/FlightManagementSystem/RouteModel.spec.js]: `ava.skip`!?  ☹️

File: test/aircraft/Pilot/clearedAsFiled.spec.js [hunk 21]
```diff
@@ -20,7 +20,8 @@ ava.afterEach(() => {
     resetNavigationLibraryFixture();
 });
 
-ava('.clearedAsFiled() grants pilot departure clearance and returns the correct response strings', (t) => {
+// TODO: Needs AircraftModel fixture, not just pilot
```
Comment (n8rzz) [inline test/aircraft/Pilot/clearedAsFiled.spec.js]: I think one of those exists already?

File: src/assets/scripts/client/InputController.js (hunk 9 above)
Comment (erikquinn) [reply src/assets/scripts/client/InputController.js]: `u2B50` is the ⭐ character, which we aren't using anymore, it's been looking for a literal asterisk. Since that's what we're using in other cases, this simply updates the character being inserted. If you're saying that a string in js will be handled differently on a mac, then I guess that's something we'll need to deal with, albeit separately from here.

File: src/assets/scripts/client/aircraft/AircraftCommander.js (hunk 10 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: affirm, I can rename.

File: src/assets/scripts/client/aircraft/AircraftCommander.js (hunk 11 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: classic case of kill-the-pilot. When we get to removing it, it would simply be `aircraft.cancelDepartureClearance()`. Doing it this ugly way for now to keep with the convention.

File: src/assets/scripts/client/aircraft/AircraftCommander.js (hunk 12 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: done

File: src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js (hunk 13 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js]:
```
Other classes will not have access to the `#_routeModel`, but can use this facade to determine whether or not a given runway assignment is allowable, given the departure procedure in the `#_routeModel`.
```
done

File: src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js (hunk 14 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js]: I guess they're kinda parallel, but I wouldn't say they're that similar... they're using different methods in like 3 different places. Plus, I can't even imagine a sensible name for a method I could abstract that out to, and my thought is that if it's too abstract to have a sensible name, it probably does more harm than good...

File: src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js (hunk 15 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js]: done

File: src/assets/scripts/client/InputController.js (hunk 9 above)
Comment (n8rzz) [reply src/assets/scripts/client/InputController.js]: ahh, ok, I'll allow it 😉

File: src/assets/scripts/client/aircraft/AircraftCommander.js (hunk 11 above)
Comment (n8rzz) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: #worksforme

File: src/assets/scripts/client/aircraft/Pilot/Pilot.js (hunk 16 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/Pilot/Pilot.js]: Here's the scenario this protects against:
1. Aircraft is filed for SID valid only from Runway 36.
1. Aircraft is successfully  cleared as filed.
1. Aircraft is taxiied to Runway 18 (without complaints, expecting to get a new SID assigned), and the pilot silently cancels their own departure clearance because they cannot accept that rwy/sid combination.
1. Aircraft will refuse takeoff clearance until receiving a routing valid for their assigned runway, or after taxiied to a valid runway and cleared-as-filed.

So this is the portion where the pilot verifies their filed route is valid for the runway they've been assigned, and if it isn't, he will reject the route, prompting the controller to assign the appropriate SID instead.

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js (hunk 17 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: Honestly, I'm not sure. Looked back and couldn't find any clues.... it's pretty apparent it's not _needed_, but at the same time I feel like there's no reason not to have it, so I'll add it back in.

Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: done

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js (hunk 18 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: done

File: src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js (hunk 19 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/aircraftCommandMap.js]: Added #919 to look into it

File: test/aircraft/FlightManagementSystem/RouteModel.spec.js (hunk 20 above)
Comment (erikquinn) [reply test/aircraft/FlightManagementSystem/RouteModel.spec.js]: done!

File: test/aircraft/Pilot/clearedAsFiled.spec.js (hunk 21 above)
Comment (erikquinn) [reply test/aircraft/Pilot/clearedAsFiled.spec.js]: It does. Due to some conflicts, i'd like to handle this test in #897 though if that's alright

File: src/assets/scripts/client/aircraft/AircraftCommander.js (hunk 10 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: done

File: src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js [hunk 22]
```diff
@@ -772,6 +820,7 @@ export default class RouteModel extends BaseModel {
         return this.currentLeg.skipToWaypointName(waypointName);
     }
 
+    // FIXME: Presume I will need to do the same thing here as with .updateStarLegForArrivalRunwayModel() ?
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js]: Is this `FIXME` still valid?

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js [hunk 23]
```diff
@@ -323,7 +305,12 @@ export default class AircraftCommandParser {
      * @return {boolean}
      */
     _isSystemCommand(callsignOrSystemCommandName) {
-        return _has(SYSTEM_COMMANDS, callsignOrSystemCommandName) &&
-            callsignOrSystemCommandName !== SYSTEM_COMMANDS.transmit;
+        const command = AIRCRAFT_COMMAND_MAP[callsignOrSystemCommandName];
```
Comment (n8rzz) [inline src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: why not use `_has()` here?

File: src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js (hunk 22 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js]: nope, removed

File: src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js (hunk 23 above)
Comment (erikquinn) [reply src/assets/scripts/client/parsers/aircraftCommandParser/AircraftCommandParser.js]: no longer needed, due to the rearrangement of the command map, and resulting adjustments to this method. should also read slightly clearer now as well, so that's also a plus!

done
